ema_pytorch/ema_pytorch.py

```diff
@@ -86,21 +92,35 @@ def __init__(
             try:
                 self.ema_model = deepcopy(model)
             except Exception as e:
-                print(f'Error: While trying to deepcopy model: {e}')
-                print('Your model was not copyable. Please make sure you are not using any LazyLinear')
+                print(f"Error: While trying to deepcopy model: {e}")
+                print(
+                    "Your model was not copyable. Please make sure you are not using any LazyLinear"
+                )
                 exit()
 
         self.ema_model.requires_grad_(False)
 
         # parameter and buffer names
 
-        self.parameter_names = {name for name, param in self.ema_model.named_parameters() if param.dtype in [torch.float, torch.float16]}
-        self.buffer_names = {name for name, buffer in self.ema_model.named_buffers() if buffer.dtype in [torch.float, torch.float16]}
```

Comment:
ah, maybe should just use `torch.is_floating_point` here. to cover complex, maybe an `or torch.is_complex` as well

Reply:
nice, if you redo the PR with just these 2 loc change as well as version bump, will accept

Reply:
I don't mind if you do the modifications